diff --git a/webkit/Source/bmalloc/bmalloc/SystemHeap.cpp b/webkit/Source/bmalloc/bmalloc/SystemHeap.cpp
--- a/webkit/Source/bmalloc/bmalloc/SystemHeap.cpp
+++ b/webkit/Source/bmalloc/bmalloc/SystemHeap.cpp
@@ -210,7 +210,13 @@ void* SystemHeap::malloc(size_t size, FailureAction action)

 void* SystemHeap::memalign(size_t alignment, size_t size, FailureAction action)
 {
-    void* result = ::aligned_alloc(alignment, size);
+    void* result = nullptr;
+#if PAS_OS(ANDROID)
+    if (::posix_memalign(&result, alignment, size))
+        result = nullptr;
+#else
+    result = ::aligned_alloc(alignment, size);
+#endif
     RELEASE_BASSERT(action == FailureAction::ReturnNull || result);
     return result;
 }
