diff --git a/webkit/Source/WTF/wtf/android/LoggingAndroid.cpp b/webkit/Source/WTF/wtf/android/LoggingAndroid.cpp
--- a/webkit/Source/WTF/wtf/android/LoggingAndroid.cpp
+++ b/webkit/Source/WTF/wtf/android/LoggingAndroid.cpp
@@ -32,20 +32,15 @@
 
 String logLevelString()
 {
-    const char* propertyValue = nullptr;
+    constexpr auto propertyName = "debug." LOG_CHANNEL_WEBKIT_SUBSYSTEM ".log";
+    char propertyValue[PROP_VALUE_MAX] { };
+    int length = __system_property_get(propertyName, propertyValue);
 
-    if (const auto* propertyInfo = __system_property_find("debug." LOG_CHANNEL_WEBKIT_SUBSYSTEM ".log")) {
-        __system_property_read_callback(propertyInfo, [](void *userData, const char*, const char* value, unsigned) {
-            auto **propertyValue = static_cast<const char**>(userData);
-            *propertyValue = value;
-        }, &propertyValue);
-    }
-
     // Disable all log channels if the property is unset or empty.
-    if (!propertyValue || !*propertyValue)
-        return makeString("-all"_s);
+    if (length <= 0)
+        return "-all"_s;
 
-    return String::fromLatin1(propertyValue);
+    return String::fromLatin1(propertyValue);
 }
 
 } // namespace WTF
