diff --git a/webkit/Source/cmake/OptionsCommon.cmake b/webkit/Source/cmake/OptionsCommon.cmake
index ea7d508967d6..b4de9ef50bbf 100644
--- a/webkit/Source/cmake/OptionsCommon.cmake
+++ b/webkit/Source/cmake/OptionsCommon.cmake
@@ -1,7 +1,11 @@
-set(CMAKE_CXX_STANDARD 23)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_CXX_EXTENSIONS OFF)
 set(CMAKE_EXPERIMENTAL_CXX_MODULE_DYNDEP OFF)
+
+# Android builds rely on downstream flags, so keep CMake from adding -O3 automatically.
+set(CMAKE_C_COMPILE_OPTIONS_RELEASE -DNDEBUG CACHE STRING "" FORCE)
+set(CMAKE_CXX_COMPILE_OPTIONS_RELEASE -DNDEBUG CACHE STRING "" FORCE)
 
 add_definitions(-DBUILDING_WITH_CMAKE=1)
 add_definitions(-DBUILDING_WEBKIT=1)
